core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/ErrorValidatorTest.scala

```diff
@@ -0,0 +1,119 @@
+package org.virtuslab.ideprobe.dependencies
+
+import scala.io.Source.fromResource
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.virtuslab.ideprobe.Config
+import org.virtuslab.ideprobe.IntelliJFixture
+import org.virtuslab.ideprobe.config.CheckConfig
+import org.virtuslab.ideprobe.config.CheckConfig.ErrorConfig
+import org.virtuslab.ideprobe.protocol.IdeMessage
+import org.virtuslab.ideprobe.protocol.IdeMessage.Level
+import org.virtuslab.ideprobe.reporting.ErrorValidator
+
+class ErrorValidatorTest {
+  private val configRoot = "probe"
+  private val defaultConfigWithErrorsEnabled: Config = Config.fromClasspath("error_validator.conf")
+  private val defaultCheckConfigWithErrorsEnabled: CheckConfig =
+    IntelliJFixture.readIdeProbeConfig(defaultConfigWithErrorsEnabled, configRoot).driver.check
+
+  private val messageBusConnectionImplNPE = fromResource("ideError_1.txt").mkString
+  private val uastMetaLanguageNPE = fromResource("ideError_2.txt").mkString
+
+  private val errorIdeMessages: List[IdeMessage] = List(messageBusConnectionImplNPE, uastMetaLanguageNPE)
+    .map(content => IdeMessage(level = Level.Error, content, None))
+
+  private val messageBusSpecificString = "at com.intellij.util.messages.impl.MessageBusConnectionImpl"
+  private val uastMetaSpecificString = "at com.intellij.uast.UastMetaLanguage"
+
+  @Test
+  def shouldIncludeAllErrorsByDefault(): Unit = {
+    val exceptions: List[Option[Exception]] = collectExceptions(defaultCheckConfigWithErrorsEnabled)
+    val exceptionsIncluded = exceptions.filter(_.nonEmpty)
+    val exceptionsIgnored = exceptions.filter(_.isEmpty)
+    assertEquals(2, exceptionsIncluded.size)
+    assertEquals(0, exceptionsIgnored.size)
+  }
+
+  @Test
+  def shouldIgnoreAllErrorsIfNotEnabled(): Unit = {
+    val checkConfigErrorsDisabled = defaultCheckConfigWithErrorsEnabled.copy(errors =
+      ErrorConfig(
```

Comment:
`defaultCheckConfigWithErrorsEnabled.errors.copy(enabled = false)` maybe

Reply:
well... Why make it more complicated by nesting another `.copy` call? IMO just `enabled = false` is much more readible

Reply:
You could have lenses to have it more concise. As they are not here yet, nested copying is unavoidable. What you did here is inlining of copy method. Why reimplementing a method that already exists. If we add more fields to ErrorConfig, this place would need to be changed. Also you clearly state that you are changing just the `enabled` property, while otherwise reader needs to think what are you putting in this `included` and `excludedMessages` (is it just the same, or does it change as well?) and this code doesn't need to be there.

Reply:
ok, done as suggested